lib/univalue.cpp

```diff
@@ -163,6 +163,9 @@ bool UniValue::findKey(const std::string& key, size_t& retIdx) const
 
 bool UniValue::checkObject(const std::map<std::string,UniValue::VType>& t)
 {
+    if (typ != VOBJ)
+        return false;
+
     size_t idx = 0;
```

Comment:
Agree with this change - but please do it outside the doxygen pull request.  Thanks!